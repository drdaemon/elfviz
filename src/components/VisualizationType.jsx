import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const VisualizationType = ({ type, typeObject, onChange }) => {
  const { name, icon } = typeObject;
  const className = cx({ active: type === name });
  return (
    <li role="presentation" className={className}>
      <a href="#" onClick={onChange.bind(this, name)}>{icon} {name}</a>
    </li>
  );
};

VisualizationType.propTypes = {
  type: PropTypes.string.isRequired,
  typeObject: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default VisualizationType;
