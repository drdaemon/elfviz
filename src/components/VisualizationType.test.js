import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import VisualizationType from './VisualizationType';

const typeObject = {
  name: 'Icicle',
  icon: 'icon',
};

describe('VisualizationType', () => {
  it('renders without crashing', () => {
    shallow(<VisualizationType type={''} typeObject={{}} onChange={() => {}} />);
  });

  it('should match an empty snapshot', () => {
    const wrapper = shallow(<VisualizationType type={''} typeObject={{}} onChange={() => {}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match a snapshot with type info', () => {
    const wrapper = shallow(<VisualizationType type={'Icicle'} typeObject={typeObject} onChange={() => {}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should trigger onChange when clicked on a element', () => {
    const onChange = jest.fn();
    const wrapper = shallow(<VisualizationType type={'Icicle2'} typeObject={typeObject} onChange={onChange} />);
    wrapper.find('a').simulate('click');
    expect(onChange).toHaveBeenCalledWith('Icicle');
  });

});
