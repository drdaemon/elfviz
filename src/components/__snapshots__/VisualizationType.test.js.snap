// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisualizationType should match a snapshot with type info 1`] = `
<li
  className="active"
  role="presentation"
>
  <a
    href="#"
    onClick={[Function]}
  >
    icon
     
    Icicle
  </a>
</li>
`;

exports[`VisualizationType should match an empty snapshot 1`] = `
<li
  className=""
  role="presentation"
>
  <a
    href="#"
    onClick={[Function]}
  >
     
  </a>
</li>
`;
